freqtrade/strategy/interface.py

```diff
@@ -152,6 +152,7 @@ def load_freqAI_model(self) -> None:
 
             self.freqai = FreqaiModelResolver.load_freqaimodel(self.config)
             self.freqai_info = self.config["freqai"]
+            self.freqai.strategy = self
```

Comment:
you'll have to find an alternative to this, as i'll not be allowing this into freqtrade this way.

Storing a reference of self in a object of self is the key for a refcount problem (refcount will never go to 0), which will lead to a memory leak over time.
while this is only relevant when using `/reload_conf` (which is supposed to recycle the strategy object completely) - it's still a smell we'll not be having.

Reply:
Agreed, I will probably just pass strategy to the environment in the environment class constructor. This was the lazy way to test proof of concept state info reinforcing in live deployments.

Reply:
Now we store dataprovider and then clean it up.